docs/feature_space_cadet.md

```diff
@@ -0,0 +1,59 @@
+# Space Cadet: The Future, Built In
+
+Steve Losh described the [Space Cadet Shift](http://stevelosh.com/blog/2012/10/a-modern-space-cadet/) quite well. Essentially, when you tap Left Shift on its own, you get an opening parenthesis; tap Right Shift on its own and you get the closing one. When held, the Shift keys function as normal. Yes, it's as cool as it sounds, and now even cooler supporting Control and Alt as well!
+
+## Usage
+
+Firstly, in your keymap, do one of the following:
+- Replace the Left Shift key with `KC_LSPO` (Left Shift, Parenthesis Open), and Right Shift with `KC_RSPC` (Right Shift, Parenthesis Close).
+- Replace the Left Control key with `KC_LCPO` (Left Control, Parenthesis Open), and Right Control with `KC_RCPC` (Right Control, Parenthesis Close).
+- Replace the Left Alt key with `KC_LAPO` (Left Alt, Parenthesis Open), and Right Alt with `KC_RAPC` (Right Alt, Parenthesis Close).
+- Replace any Shift key in your keymap with `KC_SFTENT` (Right Shift, Enter).
+
+## Keycodes
+
+|Keycode    |Description                              |
+|-----------|-----------------------------------------|
+|`KC_LSPO`  |Left Shift when held, `(` when tapped    |
+|`KC_RSPC`  |Right Shift when held, `)` when tapped   |
+|`KC_LCPO`  |Left Control when held, `(` when tapped  |
+|`KC_RCPC`  |Right Control when held, `)` when tapped |
+|`KC_LAPO`  |Left Alt when held, `(` when tapped      |
+|`KC_RAPC`  |Right Alt when held, `)` when tapped     |
+|`KC_SFTENT`|Right Shift when held, `Enter` when tapped |
+
+## Caveats
+
+Space Cadet's functionality can conflict with the default Command functionality when both Shift keys are held at the same time. Make sure that Command is disabled in your `rules.mk` with:
+
+```make
+COMMAND_ENABLE = no
+```
+
+## Configuration
+
+By default Space Cadet assumes a US ANSI layout, but if your layout uses different keys for parentheses, you can redefine them in your `config.h`. In addition, you can redefine the modifier to send on tap, or even send no modifier at all. The new configuration defines bundle all options up into a single define of 3 key codes in this order: the `Modifier` when held or when used with other keys, the `Tap Modifer` sent when tapped (no modifer if `KC_TRNS`), finally the `Keycode` sent when tapped. Now keep in mind, mods from other keys will still apply to the `Keycode` if say `KC_RSFT` is held while tapping `KC_LSPO` key with `KC_TRNS` as the `Tap Modifer`.
+
+|Define             |Default      |Description                                                          |
+|-------------------|-------------|---------------------------------------------------------------------|
+|`KC_LSPO_KEYS`     |`KC_LSFT, LSPO_MOD, LSPO_KEY`  |Send `KC_LSFT` when held, the mod and  key defined by `LSPO_MOD` and `LSPO_KEY`. |
```

Comment:
Not keen on having defines prepended with `KC_` that aren't keycodes. Also think these should be split up into individual defines eg. `_HOLD_KEY`, `_TAP_MOD`, `_TAP_KEY`. And could you please align all these tables :)

Reply:
The single line define was done to make it easier to configure as a group on a single line keeping configs smaller, cleaner, and easier to read. Otherwise you could easily have 12+ defines in your config dedicated to set this up. This is from experience of having to doing just that. Now I can definitely update these defines to reference individual defines so the user has a choice of a single config define, or a batch define if you find that more acceptable. Thoughts?

As for the KC_ prefix and table alignment, sure, will update those.

Reply:
While I do love me some aligned tables, we do not require that to accept a PR.

Reply:
Table aligned, KC_ prefix on non-keycode defines removed.
Did not add individual overrides yet, I feel it gets super cluttered, and want additional feedback.

Reply:
I can see how it simplifies the config.h, which is always good. I just think it's not as self-documenting as using individual defines. IMO it isn't immediately obvious how it works and what each part is unless you know it's meant to be three method arguments.

As you said, something like `#define LSPO_KEYS LSPO_HOLD_KEY, LSPO_TAP_MOD, LSPO_TAP_KEY` etc. would be better. That way the default would provide a bit more information on the three parts.

And yeah @skullydazed I know, I wasn't suggesting that at all, I'm just a neat freak :(

Reply:
There is something to be said for RTFM if the code is a bit confusing. As far as self documenting code goes, I get it as I tend to read code more than docs or comments. However there is something to be said about writing code that is self documenting at the cost of performance or simplicity isn't a great target. Leaving the final vote up to the maintainers.